### ISSUE TITLE
NOT_FOUND_COLUMN_IN_BLOCK prewhere VER column

### ISSUE THREAD

**Issue by den-crane** [ClickHouse/ClickHouse#45804]
```sql
CREATE TABLE mySecondReplacingMT
(
    `key` Int64,
    `someCol` String,
    `eventTime` DateTime
)
ENGINE = ReplacingMergeTree(eventTime)
ORDER BY key;

insert into mySecondReplacingMT values (123, 'test', now());

SELECT count(eventTime) FROM mySecondReplacingMT FINAL PREWHERE eventTime > now();
┌─count(eventTime)─┐
│                0 │
└──────────────────┘

SELECT count() FROM mySecondReplacingMT FINAL PREWHERE eventTime > now();

 (version 23.1.2):
Code: 10. DB::Exception: Received from localhost:9000. DB::Exception:
 Not found column eventTime in block. There are only columns: key. (NOT_FOUND_COLUMN_IN_BLOCK)

**Comment by sunny19930321** [comment]
has the problem been solved?

**Comment by den-crane** [comment]
It seems it was fixed in 24.3 by the new analyzer

https://fiddle.clickhouse.com/49fb09d4-0180-4bc8-89fd-640cb5f22f3c
